server/routes/nuxt-api/sitemaps/urls.ts

```diff
@@ -0,0 +1,46 @@
+import { defineSitemapEventHandler } from '#imports'
+import type { SitemapUrl } from '#sitemap/types'
+
+export default defineSitemapEventHandler(async (event) => {
+  const config = useRuntimeConfig()
+
+  const { type, nbSitemapSections = 1 } = getQuery(event)
+  if (!type)
+    return new Response(null, { status: 404 })
+  let { section = 1 } = getQuery(event)
+  section = parseInt(section)
+
+  const pageSize = 200
+  const selfWebUrlKey = type == 'dataservice' ? 'self_web_url' : 'page'
+
+  // computing starting api page and max page for this sitemap section
+  let currentPage = 1
+  let maxPage = null
+  await $fetch<{ path: string }[]>(`${config.public.apiBase}/api/1/${type}s/`, {
+    headers: { 'X-Fields': 'total' },
+  }).then((result) => {
+    const total = result.total
+    currentPage = Math.floor(((total / nbSitemapSections / pageSize)) * (section - 1)) + 1
+    maxPage = Math.floor(((total / nbSitemapSections / pageSize)) * section)
+    if (section == nbSitemapSections)
+      maxPage += 1 // add last incomplete page on the last sitemap
+  })
+
+  let nextPageUrl = config.public.apiBase + `/api/1/${type}s/?page_size=${pageSize}&page=${currentPage}`
```

Comment:
To improve perfs we could have the APIv2 selected for datasets and reuses?

Reply:
Done in https://github.com/datagouv/cdata/pull/428/commits/8ed103020715a68cd6cd24cb85484f269b76dc96 for datasets only